frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -55,6 +55,22 @@ if Device:setDateTime() then
     common_settings.time = {
         text = _("Time and date"),
         sub_item_table = {
+          {
+            text = _("12-hour clock"),
+            -- does the menu need to be kept open?
+            keep_menu_open = true,
+            checked_func = function()
+                return G_reader_settings:isTrue("twelve_hour_time")
```

Comment:
Since the setting isn't legacy, please also change it.

```suggestion
                return G_reader_settings:isTrue("twelve_hour_clock")
```